matrix/rules.py

```diff
@@ -340,7 +355,12 @@ def allow_event(e):
                 kind="test.schedule",
                 payload=context.suite)
         for test in context.suite:
-            await self.run_once(context, test)
+            context.test = test
+            try:
+                await self.add_model(context)
+                await self.run_once(context, test)
+            finally:
+                await self.destroy_model(context)
```

Comment:
We may want to honor a cmdline flag to bail here before cleaning the model so the developer can check the remaining units in more detail